Added Features to Country Map Viz Type

avrilpearl opened the pull request:
I have added more functionality to the Country Map Visualization type, specific to standard Choropleth map generation in the Geography world:

- the ability to specify the scale for the choropleth map (linear, quantize, quantile) <img width="231" alt="screen shot 2017-09-01 at 2 30 51 pm" src="https://user-images.githubusercontent.com/9359261/29983144-80055c4c-8f22-11e7-8638-900cf8decad0.png">
- the ability to specify the number of buckets to group your data in (3-12) <img width="225" alt="screen shot 2017-09-01 at 2 34 06 pm" src="https://user-images.githubusercontent.com/9359261/29983165-a879f796-8f22-11e7-9f6c-981a122d9a84.png">
- the ability to display a legend <img width="75" alt="screen shot 2017-09-01 at 2 35 40 pm" src="https://user-images.githubusercontent.com/9359261/29983243-ee6260cc-8f22-11e7-9b84-fccaffd0a5d4.png"> <img width="495" alt="screen shot 2017-09-01 at 2 35 49 pm" src="https://user-images.githubusercontent.com/9359261/29983230-e10f2162-8f22-11e7-86dc-61b9c653ee37.png">
- the ability to specify the numerical format of your data for display in the hover feature and the legend 
<img width="172" alt="screen shot 2017-09-01 at 2 38 08 pm" src="https://user-images.githubusercontent.com/9359261/29983331-50100da6-8f23-11e7-8d95-28d36c298a9b.png"> <img width="447" alt="screen shot 2017-09-01 at 2 37 57 pm" src="https://user-images.githubusercontent.com/9359261/29983319-45c2f016-8f23-11e7-9247-37aa86085962.png">
- a broader set of color shemes to pick from based on the mapping scales from [colorbrewer](http://colorbrewer2.org/#type=sequential&scheme=BuGn&n=3) and made sure all values have a border because now the color scheme consists of more than just 'white' for lighter values   
<img width="450" alt="screen shot 2017-09-01 at 2 38 17 pm" src="https://user-images.githubusercontent.com/9359261/29983356-74c7f37a-8f23-11e7-87b9-137799025664.png">

_Note_ : As the linear color scheme also impacts the heatmap viz, the added color schemes are available in the selector for that viz type.

avrilpearl committed:
Added massachusetts geojson

in ma state plane projects

avrilpearl committed:
Added massachusetts geojson

in ma state plane projects

avrilpearl committed:
added massachusetts to select list

avrilpearl committed:
added massachusetts to select list

avrilpearl committed:
updated airbnb color swatches

avrilpearl committed:
add overflow scroll to calendar heatmap css

avrilpearl committed:
add overflow scroll to calendar heatmap css

avrilpearl committed:
added 'flask==0.12.1'

avrilpearl committed:
added 'flask==0.12.1'

avrilpearl committed:
added colorbrewer to required packages

avrilpearl committed:
added colorbrewer to required packages

avrilpearl committed:
added colorbrewer import statement

avrilpearl committed:
added colorbrewer import statement

avrilpearl committed:
added d3-scale-chromatic to package.json

avrilpearl committed:
added d3-scale-chromatic to package.json

avrilpearl committed:
attempt to add color chromatic scale to colors

avrilpearl committed:
attempt to add color chromatic scale to colors

avrilpearl committed:
added d3-scale

avrilpearl committed:
added d3-scale

avrilpearl committed:
Update colors.js

avrilpearl committed:
Update package.json

avrilpearl committed:
Update package.json

avrilpearl committed:
Update colors.js

avrilpearl committed:
Update colors.js

avrilpearl committed:
added overflow scroll

avrilpearl committed:
added overflow scroll

avrilpearl committed:
changed subheader formatting

avrilpearl committed:
changed subheader formatting

avrilpearl committed:
added YlGnBu to color spectrums

avrilpearl committed:
added YlGnBu to color spectrums

avrilpearl committed:
added yellow green blue to selector

avrilpearl committed:
added yellow green blue to selector

avrilpearl committed:
added yellow green blue to color selector

avrilpearl committed:
added yellow green blue to color selector

avrilpearl committed:
added psycop2 to setup.py

avrilpearl committed:
added psycop2 to setup.py

avrilpearl committed:
added d3-scale-chromatic

avrilpearl committed:
added d3-scale-chromatic

avrilpearl committed:
change map to d3.scale.quantile instead of linear

avrilpearl committed:
change map to d3.scale.quantile instead of linear

avrilpearl committed:
added logic to handle non map scaling

avrilpearl committed:
added 'map' to color scaler function

try to test out color scaler with quantile for the map

avrilpearl committed:
added 'map' to color scaler function

try to test out color scaler with quantile for the map

avrilpearl committed:
Merge pull request #1 from massgov/country-map-styling

Country map styling

avrilpearl committed:
check for val

avrilpearl committed:
edit color scaler variable

avrilpearl committed:
edit color scaler variable

avrilpearl committed:
updated quantile scale input

avrilpearl committed:
added console msgs

avrilpearl committed:
Update colors.js

avrilpearl committed:
updated to pull correct values

avrilpearl committed:
updated to pull correct values

avrilpearl committed:
debugging

avrilpearl committed:
Update common.js

avrilpearl committed:
Update common.js

avrilpearl committed:
rolled back comment

avrilpearl committed:
rolled back comment

avrilpearl committed:
added colors log

avrilpearl committed:
added colors log

avrilpearl committed:
added data log

avrilpearl committed:
added data array for country

avrilpearl committed:
added console log of input accessor

avrilpearl committed:
added console log of input accessor

avrilpearl committed:
added domain for quantile

avrilpearl committed:
added domain for quantile

avrilpearl committed:
update to domain data.map

avrilpearl committed:
update to domain data.map

avrilpearl committed:
updated domain again

avrilpearl committed:
updated domain again

avrilpearl committed:
removed element from array

avrilpearl committed:
added more elements to color array - testing len issue

avrilpearl committed:
added more elements to color array - testing len issue

avrilpearl committed:
working on legend for map

avrilpearl committed:
working on legend for map

avrilpearl committed:
added fd.scale_color_scheme to color function

avrilpearl committed:
added fd.scale_color_scheme to color function

avrilpearl committed:
Update country_map.js

avrilpearl committed:
Update country_map.js

avrilpearl committed:
Merge remote-tracking branch 'apache/master'

# Conflicts:
#	superset/assets/javascripts/modules/colors.js

avrilpearl committed:
Updated colors.js to resolve merge conflicts

avrilpearl committed:
Updated colors.js to resolve merge conflicts

avrilpearl committed:
resolved merge conflicts with country map

avrilpearl committed:
resolved merge conflicts with country map

avrilpearl committed:
Merge branch 'master' of github.com:massgov/incubator-superset

Udpated master branch based on upstream apache superset source.

avrilpearl committed:
Added legend option to country map

Added the option to include a legend in the country map.
Added number formatting option to the country map (impacts hover metric
and legend display)
Updated continuous d3.scale in the colors.js module so that scale
represents true dataset range.

avrilpearl committed:
Removed non-related to legend addition changes commited

Removed all committed changes that were not-related to the addition of
a legend to the country map visualization.

avrilpearl committed:
added scale and bucket controls to map

avrilpearl committed:
update based on controls.jsx color scheme changes

avrilpearl committed:
changes

avrilpearl committed:
changes

avrilpearl committed:
changes

avrilpearl committed:
changes

avrilpearl committed:
default value assigned to color scaler factory

avrilpearl committed:
added color brewer scales to linear color scheme options

added color brewer scales to linear color scheme option list
updated dropdown with swatches to have grey border and show all colors

avrilpearl committed:
added border to linear color scheme swatches dropdown

avrilpearl committed:
Merge remote-tracking branch 'apache/master' into add-legend-to-country-map

# Conflicts:
#	superset/assets/javascripts/modules/colors.js

avrilpearl committed:
backend sync son

avrilpearl committed:
updated show legend logic for country map

avrilpearl committed:
Merge remote-tracking branch 'apache/master' into add-legend-to-country-map

mistercrunch commented on superset/assets/javascripts/SqlLab/components/SplitPane.jsx:
```diff
@@ -0,0 +1,146 @@
+import React from 'react';
```
I don't think this file belongs in this PR

avrilpearl replied:
@mistercrunch i have removed this from the branch and pr

mistercrunch reviewed (changes requested):
Remove unrelated changes

avrilpearl committed:
Merge branch 'apache/master' into country-map-legend-bucketing-colorschemes

avrilpearl committed:
fixed legend checkbox control

fixed the checkbox control for the map legend display

avrilpearl committed:
removed SplitPane.jsx from branch

per comment of @mistercrunch - removed the misplaced SplitPane.jsx from
this branch and pr-3408

avrilpearl committed:
removed unused colorScalerFactory from ColorSchemeControl.jsx

avrilpearl committed:
syntax fixes to ColorSchemeControl.jsx

avrilpearl committed:
fix to syntax travis warnings in colors.js and controls.jsx

avrilpearl committed:
fixed syntax and build errors

avrilpearl committed:
fixed es-lint syntax issues

avrilpearl committed:
fixed all es-lint syntax errors

avrilpearl committed:
implemented default scale type in colorScalerFactory function

solved issue with heatmap.js error as scale is not an option for this
viz type.

avrilpearl committed:
scale selector default value fixed, map legend fix

- fixed the typo related to the default value for the scale selector
control.
- fixed how the legend is added to the map as to not cause issues on
the dashboard if two country map slices are present.

slarrain commented:
Do you think this is ready to be reviewed @mistercrunch ?

xrmx commented:
@slarrain quite hard to review while the PR has merge conflicts

slarrain commented:
@xrmx yep, you're right Ricardo. They arose with the new version. Sorry about that.

wildanm commented:
Is there any plan to merge/add this feature to the master ?
